Assets/MRTK/SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs

```diff
@@ -425,8 +425,10 @@ void IMixedRealityTouchHandler.OnTouchCompleted(HandTrackingInputEventData event
         /// </summary>
         /// 
         public Vector3 GetWorldPositionAlongPushDirection(float localDistance)
-        {
-            float distance = (distanceSpaceMode == SpaceMode.Local) ? localDistance * LocalToWorldScale : localDistance;
+        {   
+            // We clamp the calculated local distance to the maxPushDistance,
+            // because LocalToWorldScale may return NaNs when object scale is zero.
+            float distance = Mathf.Clamp(0f, maxPushDistance, (distanceSpaceMode == SpaceMode.Local) ? localDistance * LocalToWorldScale : localDistance);
```

Comment:
Also something I don't know too much about (maybe @thalbern @CDiaz-MS @cre8ivepark know) - should the min distance here be the starting push distance instead (i.e. is that actually the lower clamp value?) The two happen to match by default (i.e. both zero), the only reason I ask is because usually when there's a min and max value defined elsewhere and we clamp somewhere in the code using only one of them, there's probably a good reason for us not using that (which can be worth a comment)

Reply:
@wiwei There is actually an alternative approach here I'm going to pursue; the issue is that the NaNs are actually working their way into more things than just this one distance calculation.